python/private/proto/py_proto_library.bzl

```diff
@@ -25,16 +26,52 @@ _PyProtoInfo = provider(
         "imports": """
             (depset[str]) The field forwarding PyInfo.imports coming from
             the proto language runtime dependency.""",
-        "runfiles_from_proto_deps": """
-            (depset[File]) Files from the transitive closure implicit proto
-            dependencies""",
         "transitive_sources": """(depset[File]) The Python sources.""",
     },
 )
 
 def _filter_provider(provider, *attrs):
     return [dep[provider] for attr in attrs for dep in attr if provider in dep]
 
+def _get_import_path(ctx, proto_info):
```

Comment:
Can you explain the need for this a bit, please?

The issue it seems to be trying to solve is making sure the generated .py files end up on sys.path, and thus actually importable. Makes sense, we need code to be imported :).

However, I see two issues:

If I'm reading this right, it's adding the directory that has the proto source files to sys.path? This essentially turns every proto into a top-level import. This won't work for a couple reasons: it'll explode sys.path (which is already space constrained) and it'll create a conflict if different protos have the same name.

More generally, though, we have the same problem here as with py_library: the correct/desired sys.path entry can't be automatically determined based on the file path. As an example, the code may be homed in e.g. "src/whatever/protos", and the intended import is "import whatever.protos".

So, really, I don't think we can generate any sys.path information here automatically as part of py_proto_library.

We could add an `imports` attribute to `py_proto_library` that behaves as `py_library.imports` does, though.